executor/src/witgen/mod.rs

```diff
@@ -238,18 +238,16 @@ impl<'a, T> FixedColumn<'a, T> {
     }
 }
 
-#[derive(Debug)]
-pub struct Query<'a, T> {
-    /// The query expression
-    expr: &'a Expression<T>,
-    /// The polynomial that is referenced by the query
-    poly: PolynomialReference,
-}
-
 #[derive(Debug)]
 pub struct WitnessColumn<'a, T> {
-    name: String,
-    query: Option<Query<'a, T>>,
+    /// A polynomial reference that points to this column in the "current" row.
```

Comment:
I have the impression this comment is more confusing than helpful. It is just a PolynomialReference for this exact column, isn't it?

Reply:
The confusing part I'm trying to explain is that `PolynomialReference` also has a `next` flag, which only makes sense in the context of a concrete row.

I feel like this is a bit of a hack, because the `EvalValue` type that's used everywhere in the processors uses `&'a PolynomialReference` as a key, so we have to prepare those polynomial references here and store it in `FixedData`, so that `self.fixed_data.witness_cols[poly_id].poly` has the right lifetime.

Reply:
Oh, I see! Then maybe write "`next` is asserted to be false" or something. I mean we could also use a PolyID, but then we would not be able to fill the name if that's needed.

Reply:
I added an "(i.e., the "next" flag is set to false)" (there isn't really an assertion).